tree-sitter-python/test/Main.hs

```diff
@@ -15,13 +15,15 @@ import           TreeSitter.Python
 import qualified TreeSitter.Python.AST as Py
 import           TreeSitter.Unmarshal
 
+-- TODO: add tests that verify correctness for product, sum and leaf types
```

Comment:
separately, I think I should add some improved tests in a follow-up PR to validate the main structural differences.